src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useState } from "react";
+import { ToDo } from "../../types";
+
+type Props = {
+  onAdd: (todo: ToDo) => void,
+  todos: ToDo[],
+  users: ToDo['user'][]
+}
+
+const getLastToDoId = (todos: ToDo[]): number => {
+  return todos.reduce((maxId, todo: ToDo) => {
+    if (todo.id > maxId) {
+      return todo.id;
+    }
+
+    return maxId
+  }, 0)
+}
+
+export const TodoForm: React.FC<Props> = ({
+  onAdd,
+  todos,
+  users
+}) => {
+  const [title, setTitle] = useState('');
+  const [isTitleFailed, setIsTitleFailed] = useState(false);
+
+  const [selectedUserId, setSelectedUserId] = useState(0)
+  const [isSelectedUserIdFailed, setSelectedUserIdFailed] = useState(false);
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    let failed = false;
+
+    if (title === '') {
+      failed = true;
+      setIsTitleFailed(true);
+    }
```

Comment:
```suggestion
    if (!title) {
      setIsTitleFailed(true);
      return;
    }
```
Don't compare it with an empty string, and the same with `selectedUserId === 0`. Fix it
https://developer.mozilla.org/en-US/docs/Glossary/Falsy